services/workflows-service/src/case-management/controllers/case-management.controller.ts

```diff
@@ -96,13 +106,26 @@ export class CaseManagementController {
           activeMonitorings: true,
           updatedAt: true,
         },
+        where: {
+          Counterparty: {
+            every: {
+              endUserId: null,
+            },
+          },
+        },
         take: searchQueryParams.page.size,
         skip: (searchQueryParams.page.number - 1) * searchQueryParams.page.size,
       },
       [projectId],
     );
     const typedEndUsers = endUsers as Array<
       (typeof endUsers)[number] & {
+        endUsersOnBusinesses: Array<{
+          position: string[];
+          business: {
+            companyName: string;
+          };
+        }>;
         businesses: Array<{
           companyName: string;
         }>;
```

Comment:
Do you mind using `Pick` on the Prisma types of `Business` and `EndUsersOnBusinesses`?